src/Sulu/Bundle/LocationBundle/Geolocator/Service/NominatimGeolocator.php

```diff
@@ -94,33 +95,51 @@ public function locate(string $query): GeolocatorResponse
 
         if ($response instanceof ResponseInterface) {
             // BC to support for guzzle client
-            $results = \json_decode($response->getBody(), true);
+            $responseBody = \json_decode($response->getBody(), true);
         } else {
-            $results = $response->toArray();
+            $responseBody = $response->toArray();
         }
 
         $response = new GeolocatorResponse();
-        foreach ($results as $result) {
-            $location = new GeolocatorLocation();
-
-            foreach ([
-                'setStreet' => 'road',
-                'setNumber' => 'house_number',
-                'setCode' => 'postcode',
-                'setTown' => 'city',
-                'setCountry' => 'country_code',
-            ] as $method => $key) {
-                if (isset($result['address'][$key])) {
-                    $location->$method($result['address'][$key]);
+
+        if (0 === $responseBody['info']['statuscode']) {
```

Comment:
use early returns instead of nested if else statement:

```php
if (0 !== ($responseBody['info']['statuscode'] ?? null)) {
    throw new Exception($responseBody['info']['messages'][0] ?? 'Unexpected error');
}

if (!\array_key_exists('results', (array) $responseBody) and 0 < \count($responseBody['results'])) {
    throw new Exception('No results found.');
}
```

We also can not be 100% if the keys exists or go with `??` a default value.